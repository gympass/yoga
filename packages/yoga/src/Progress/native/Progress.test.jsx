import React from 'react';
import { render } from '@testing-library/react-native';

import { ThemeProvider, Progress } from '../..';

describe('<Progress />', () => {
  it('should match snapshot', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Progress />
        <Progress max={100} />
        <Progress max={100} value={60} />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot with label string', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Progress max={100} label={{ value: 'Some decription here' }} />
        <Progress
          max={100}
          value={65}
          label={{ value: 'Some decription here', placement: 'right' }}
        />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot with label number', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Progress max={100} value={20} label={{ value: 20 }} />
        <Progress
          max={100}
          value={65}
          label={{ value: 65, placement: 'right' }}
        />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot with all variants', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Progress
          max={100}
          value={20}
          label={{ value: 20 }}
          variant="primary"
        />
        <Progress
          max={100}
          value={20}
          label={{ value: 20 }}
          variant="secondary"
        />
        <Progress max={100} value={20} label={{ value: 20 }} variant="vibin" />
        <Progress max={100} value={20} label={{ value: 20 }} variant="hope" />
        <Progress max={100} value={20} label={{ value: 20 }} variant="energy" />
        <Progress max={100} value={20} label={{ value: 20 }} variant="relax" />
        <Progress max={100} value={20} label={{ value: 20 }} variant="peace" />
        <Progress max={100} value={20} label={{ value: 20 }} variant="verve" />
        <Progress max={100} value={20} label={{ value: 20 }} variant="uplift" />
        <Progress
          max={100}
          value={20}
          label={{ value: 20 }}
          variant="deepPurple"
        />
        <Progress
          max={100}
          value={20}
          label={{ value: 20 }}
          variant="stamina"
        />
        <Progress max={100} value={20} label={{ value: 20 }} variant="dark" />
        <Progress max={100} value={20} label={{ value: 20 }} variant="medium" />
        <Progress max={100} value={20} label={{ value: 20 }} variant="deep" />
        <Progress max={100} value={20} label={{ value: 20 }} variant="light" />
        <Progress max={100} value={20} label={{ value: 20 }} variant="clear" />
        <Progress max={100} value={20} label={{ value: 20 }} variant="white" />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot with positions prop system', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Progress position="absolute" top="small" left="small" />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot with spacing prop system', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Progress m="small" padding="medium" />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot with background color prop system', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Progress backgroundColor="primary" />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot with border prop system', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Progress b="small" bRadius="small" />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot with elevation prop system', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Progress elevation="small" />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
