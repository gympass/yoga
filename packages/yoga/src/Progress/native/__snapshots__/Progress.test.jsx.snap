// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Progress /> should match snapshot 1`] = `
Array [
  <View
    align="left"
    isNumber={false}
    style={
      Array [
        Object {
          "flexDirection": "column",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#CA4808",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "NaN%",
            },
          ]
        }
        variant="verve"
        width={NaN}
      />
    </View>
  </View>,
  <View
    align="left"
    isNumber={false}
    style={
      Array [
        Object {
          "flexDirection": "column",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#CA4808",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "NaN%",
            },
          ]
        }
        variant="verve"
        width={NaN}
      />
    </View>
  </View>,
  <View
    align="left"
    isNumber={false}
    style={
      Array [
        Object {
          "flexDirection": "column",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#CA4808",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "60%",
            },
          ]
        }
        variant="verve"
        width={60}
      />
    </View>
  </View>,
]
`;

exports[`<Progress /> should match snapshot with all variants 1`] = `
Array [
  <View
    align="left"
    isNumber={true}
    style={
      Array [
        Object {
          "flexDirection": "row-reverse",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#D8385E",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "20%",
            },
          ]
        }
        variant="primary"
        width={20}
      />
    </View>
    <Text
      align="left"
      isNumber={true}
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "marginRight": 8,
            "textAlign": "left",
            "width": 22,
          },
        ]
      }
    >
      20
    </Text>
  </View>,
  <View
    align="left"
    isNumber={true}
    style={
      Array [
        Object {
          "flexDirection": "row-reverse",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#231B22",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "20%",
            },
          ]
        }
        variant="secondary"
        width={20}
      />
    </View>
    <Text
      align="left"
      isNumber={true}
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "marginRight": 8,
            "textAlign": "left",
            "width": 22,
          },
        ]
      }
    >
      20
    </Text>
  </View>,
  <View
    align="left"
    isNumber={true}
    style={
      Array [
        Object {
          "flexDirection": "row-reverse",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#D8385E",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "20%",
            },
          ]
        }
        variant="vibin"
        width={20}
      />
    </View>
    <Text
      align="left"
      isNumber={true}
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "marginRight": 8,
            "textAlign": "left",
            "width": 22,
          },
        ]
      }
    >
      20
    </Text>
  </View>,
  <View
    align="left"
    isNumber={true}
    style={
      Array [
        Object {
          "flexDirection": "row-reverse",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#1D856C",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "20%",
            },
          ]
        }
        variant="hope"
        width={20}
      />
    </View>
    <Text
      align="left"
      isNumber={true}
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "marginRight": 8,
            "textAlign": "left",
            "width": 22,
          },
        ]
      }
    >
      20
    </Text>
  </View>,
  <View
    align="left"
    isNumber={true}
    style={
      Array [
        Object {
          "flexDirection": "row-reverse",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFC24C",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "20%",
            },
          ]
        }
        variant="energy"
        width={20}
      />
    </View>
    <Text
      align="left"
      isNumber={true}
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "marginRight": 8,
            "textAlign": "left",
            "width": 22,
          },
        ]
      }
    >
      20
    </Text>
  </View>,
  <View
    align="left"
    isNumber={true}
    style={
      Array [
        Object {
          "flexDirection": "row-reverse",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#5340C9",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "20%",
            },
          ]
        }
        variant="relax"
        width={20}
      />
    </View>
    <Text
      align="left"
      isNumber={true}
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "marginRight": 8,
            "textAlign": "left",
            "width": 22,
          },
        ]
      }
    >
      20
    </Text>
  </View>,
  <View
    align="left"
    isNumber={true}
    style={
      Array [
        Object {
          "flexDirection": "row-reverse",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFB0A7",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "20%",
            },
          ]
        }
        variant="peace"
        width={20}
      />
    </View>
    <Text
      align="left"
      isNumber={true}
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "marginRight": 8,
            "textAlign": "left",
            "width": 22,
          },
        ]
      }
    >
      20
    </Text>
  </View>,
  <View
    align="left"
    isNumber={true}
    style={
      Array [
        Object {
          "flexDirection": "row-reverse",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#CA4808",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "20%",
            },
          ]
        }
        variant="verve"
        width={20}
      />
    </View>
    <Text
      align="left"
      isNumber={true}
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "marginRight": 8,
            "textAlign": "left",
            "width": 22,
          },
        ]
      }
    >
      20
    </Text>
  </View>,
  <View
    align="left"
    isNumber={true}
    style={
      Array [
        Object {
          "flexDirection": "row-reverse",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FF4438",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "20%",
            },
          ]
        }
        variant="uplift"
        width={20}
      />
    </View>
    <Text
      align="left"
      isNumber={true}
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "marginRight": 8,
            "textAlign": "left",
            "width": 22,
          },
        ]
      }
    >
      20
    </Text>
  </View>,
  <View
    align="left"
    isNumber={true}
    style={
      Array [
        Object {
          "flexDirection": "row-reverse",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#710252",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "20%",
            },
          ]
        }
        variant="deepPurple"
        width={20}
      />
    </View>
    <Text
      align="left"
      isNumber={true}
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "marginRight": 8,
            "textAlign": "left",
            "width": 22,
          },
        ]
      }
    >
      20
    </Text>
  </View>,
  <View
    align="left"
    isNumber={true}
    style={
      Array [
        Object {
          "flexDirection": "row-reverse",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#231B22",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "20%",
            },
          ]
        }
        variant="stamina"
        width={20}
      />
    </View>
    <Text
      align="left"
      isNumber={true}
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "marginRight": 8,
            "textAlign": "left",
            "width": 22,
          },
        ]
      }
    >
      20
    </Text>
  </View>,
  <View
    align="left"
    isNumber={true}
    style={
      Array [
        Object {
          "flexDirection": "row-reverse",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": undefined,
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "20%",
            },
          ]
        }
        variant="dark"
        width={20}
      />
    </View>
    <Text
      align="left"
      isNumber={true}
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "marginRight": 8,
            "textAlign": "left",
            "width": 22,
          },
        ]
      }
    >
      20
    </Text>
  </View>,
  <View
    align="left"
    isNumber={true}
    style={
      Array [
        Object {
          "flexDirection": "row-reverse",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#9898A6",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "20%",
            },
          ]
        }
        variant="medium"
        width={20}
      />
    </View>
    <Text
      align="left"
      isNumber={true}
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "marginRight": 8,
            "textAlign": "left",
            "width": 22,
          },
        ]
      }
    >
      20
    </Text>
  </View>,
  <View
    align="left"
    isNumber={true}
    style={
      Array [
        Object {
          "flexDirection": "row-reverse",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#6B6B78",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "20%",
            },
          ]
        }
        variant="deep"
        width={20}
      />
    </View>
    <Text
      align="left"
      isNumber={true}
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "marginRight": 8,
            "textAlign": "left",
            "width": 22,
          },
        ]
      }
    >
      20
    </Text>
  </View>,
  <View
    align="left"
    isNumber={true}
    style={
      Array [
        Object {
          "flexDirection": "row-reverse",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#D7D7E0",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "20%",
            },
          ]
        }
        variant="light"
        width={20}
      />
    </View>
    <Text
      align="left"
      isNumber={true}
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "marginRight": 8,
            "textAlign": "left",
            "width": 22,
          },
        ]
      }
    >
      20
    </Text>
  </View>,
  <View
    align="left"
    isNumber={true}
    style={
      Array [
        Object {
          "flexDirection": "row-reverse",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#F5F5FA",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "20%",
            },
          ]
        }
        variant="clear"
        width={20}
      />
    </View>
    <Text
      align="left"
      isNumber={true}
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "marginRight": 8,
            "textAlign": "left",
            "width": 22,
          },
        ]
      }
    >
      20
    </Text>
  </View>,
  <View
    align="left"
    isNumber={true}
    style={
      Array [
        Object {
          "flexDirection": "row-reverse",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "20%",
            },
          ]
        }
        variant="white"
        width={20}
      />
    </View>
    <Text
      align="left"
      isNumber={true}
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "marginRight": 8,
            "textAlign": "left",
            "width": 22,
          },
        ]
      }
    >
      20
    </Text>
  </View>,
]
`;

exports[`<Progress /> should match snapshot with background color prop system 1`] = `
<View
  align="left"
  backgroundColor="primary"
  isNumber={false}
  style={
    Array [
      Object {
        "backgroundColor": "#D8385E",
        "flexDirection": "column",
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F5F5FA",
          "borderRadius": 8,
          "flexGrow": 1,
          "height": 8,
          "marginBottom": "auto",
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": "auto",
          "position": "relative",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#CA4808",
            "borderRadius": 8,
            "height": "100%",
            "position": "absolute",
            "width": "NaN%",
          },
        ]
      }
      variant="verve"
      width={NaN}
    />
  </View>
</View>
`;

exports[`<Progress /> should match snapshot with border prop system 1`] = `
<View
  align="left"
  b="small"
  bRadius="small"
  isNumber={false}
  style={
    Array [
      Object {
        "borderColor": "black",
        "borderRadius": 8,
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexDirection": "column",
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F5F5FA",
          "borderRadius": 8,
          "flexGrow": 1,
          "height": 8,
          "marginBottom": "auto",
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": "auto",
          "position": "relative",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#CA4808",
            "borderRadius": 8,
            "height": "100%",
            "position": "absolute",
            "width": "NaN%",
          },
        ]
      }
      variant="verve"
      width={NaN}
    />
  </View>
</View>
`;

exports[`<Progress /> should match snapshot with elevation prop system 1`] = `
<View
  align="left"
  elevation="small"
  isNumber={false}
  style={
    Array [
      Object {
        "flexDirection": "column",
        "shadowColor": "rgba(152, 152, 166, 0.2)",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 4,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F5F5FA",
          "borderRadius": 8,
          "flexGrow": 1,
          "height": 8,
          "marginBottom": "auto",
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": "auto",
          "position": "relative",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#CA4808",
            "borderRadius": 8,
            "height": "100%",
            "position": "absolute",
            "width": "NaN%",
          },
        ]
      }
      variant="verve"
      width={NaN}
    />
  </View>
</View>
`;

exports[`<Progress /> should match snapshot with label number 1`] = `
Array [
  <View
    align="left"
    isNumber={true}
    style={
      Array [
        Object {
          "flexDirection": "row-reverse",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#CA4808",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "20%",
            },
          ]
        }
        variant="verve"
        width={20}
      />
    </View>
    <Text
      align="left"
      isNumber={true}
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "marginRight": 8,
            "textAlign": "left",
            "width": 22,
          },
        ]
      }
    >
      20
    </Text>
  </View>,
  <View
    align="right"
    isNumber={true}
    style={
      Array [
        Object {
          "flexDirection": "row",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#CA4808",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "65%",
            },
          ]
        }
        variant="verve"
        width={65}
      />
    </View>
    <Text
      align="right"
      isNumber={true}
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "marginLeft": 8,
            "textAlign": "right",
            "width": 22,
          },
        ]
      }
    >
      65
    </Text>
  </View>,
]
`;

exports[`<Progress /> should match snapshot with label string 1`] = `
Array [
  <View
    align="left"
    isNumber={false}
    style={
      Array [
        Object {
          "flexDirection": "column",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#CA4808",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "NaN%",
            },
          ]
        }
        variant="verve"
        width={NaN}
      />
    </View>
    <Text
      align="left"
      isNumber={false}
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "marginTop": 4,
            "textAlign": "left",
          },
        ]
      }
    >
      Some decription here
    </Text>
  </View>,
  <View
    align="right"
    isNumber={false}
    style={
      Array [
        Object {
          "flexDirection": "column",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F5F5FA",
            "borderRadius": 8,
            "flexGrow": 1,
            "height": 8,
            "marginBottom": "auto",
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": "auto",
            "position": "relative",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#CA4808",
              "borderRadius": 8,
              "height": "100%",
              "position": "absolute",
              "width": "65%",
            },
          ]
        }
        variant="verve"
        width={65}
      />
    </View>
    <Text
      align="right"
      isNumber={false}
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "marginTop": 4,
            "textAlign": "right",
          },
        ]
      }
    >
      Some decription here
    </Text>
  </View>,
]
`;

exports[`<Progress /> should match snapshot with positions prop system 1`] = `
<View
  align="left"
  isNumber={false}
  left="small"
  position="absolute"
  style={
    Array [
      Object {
        "flexDirection": "column",
        "left": 16,
        "position": "absolute",
        "top": 16,
        "width": "100%",
      },
    ]
  }
  top="small"
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F5F5FA",
          "borderRadius": 8,
          "flexGrow": 1,
          "height": 8,
          "marginBottom": "auto",
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": "auto",
          "position": "relative",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#CA4808",
            "borderRadius": 8,
            "height": "100%",
            "position": "absolute",
            "width": "NaN%",
          },
        ]
      }
      variant="verve"
      width={NaN}
    />
  </View>
</View>
`;

exports[`<Progress /> should match snapshot with spacing prop system 1`] = `
<View
  align="left"
  isNumber={false}
  m="small"
  padding="medium"
  style={
    Array [
      Object {
        "flexDirection": "column",
        "marginBottom": 16,
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 16,
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F5F5FA",
          "borderRadius": 8,
          "flexGrow": 1,
          "height": 8,
          "marginBottom": "auto",
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": "auto",
          "position": "relative",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#CA4808",
            "borderRadius": 8,
            "height": "100%",
            "position": "absolute",
            "width": "NaN%",
          },
        ]
      }
      variant="verve"
      width={NaN}
    />
  </View>
</View>
`;
